### DISCUSSION TITLE
Financial benefits for DevPool partners

### DISCUSSION THREAD

**Discussion by sergfeldman** [ubiquity/.github#11] [DevPool]
One of the directions of the DevPool evolution could be  financial benefits for the partner Web3 projects.

For example, Self-paid bounties - Web3 projects can get their bounties resolved for free.
Revenue sources that cover cost of bounties:

1. Put the overall Bounty budget into the Ubiquity vault (that aggregates strategies like LP ETH-USDC).
Revenue from the vault will be used to pay for bounties.

2. Educate developers-trainees - they solve issues under supervision of experienced developers and pay fee.
Fee is shared between experienced developers and the Web3 project that provided the bounty.

**Comment by 0x4007** [comment]
A small semi related comment but I am leaning towards referring to all of this as the DevPool program (or just the DevPool) because of the various comments I received on how this spans beyond just bounties.

> Put the overall Bounty budget into the Ubiquity vault (that aggregates strategies like LP ETH-USDC).
Revenue from the vault will be used to pay for bounties.

We discussed a similar point before but I would argue that if we have the DevPool partners mint Ubiquity Dollars, then thats "cash in the bank" that we should be deploying as part of our normal collateral strategies for example, the straddle (call/put option) plus ETH LP strategy to get 10-20% in yield in collateral.

In this case we'd only be on the hook to cash out when people redeem their Ubiquity Dollars for the underlying collateral (I think this still needs some more thought because ideally we should have some type of "rollup" to withdraw active deployed strategies. Either that or we pass along the gas fee for an instant withdrawal/cashout. DYDX has a similar model which I think works quite well. 

>  financial benefits for Bounty program partners

Ultimately I am a bit hesitant to be giving out cash rewards (unless we clearly write these off as temporary marketing/growth expenses) because the benefits this product/service provides should hopefully be enough to get profits from partners. Although we have a healthy runway still, we should have urgency to start making profit off of products/services as soon as possible.

**Reply by sergfeldman** [reply]
> I would argue that if we have the DevPool partners mint Ubiquity Dollars, then thats "cash in the bank" that we should be deploying as part of our normal collateral strategies 

This proposal is to create strong financial benefits to the DevPool partners so that they store in Ubiquity Dollars (or Ubiquity vault) not only operational budget (ie 2K uAD for the currebtly open bounties), but the overall bounty budget (ie 30K uAD for the upcoming year bounties)

**Reply by sergfeldman** [reply]
> I think this still needs some more thought because ideally we should have some type of "rollup" to withdraw active deployed strategies. Either that or we pass along the gas fee for an instant withdrawal/cashout.

Compensation of gas fees when DevPool partners pay out bounties can be one of the financial benefits for them.

Although, they will pay fee for minting uAD.
These spends should be calculated to provide the clear proposal to partners.

**Reply by sergfeldman** [reply]
> Although we have a healthy runway still, we should have urgency to start making profit off of products/services as soon as possible.

I fully support this!
Financial benefits for DevPool partners should be created NOT at the expense of Ubiquity.

In the given examples
1. Revenue will be increased due to increased liquidity provided by partners.
2. Revenue will come from trainees (based on Nate's idea).

**Reply by 0x4007** [reply]
> but the overall bounty budget (ie 30K uAD for the upcoming year bounties)

We could provide discounts for bulk buys. We just need to do some rough calculations e.g. we can get 10% APR on $50k so we should give a 5% discount if we think it'll take them a year to spend it...or something like that. Still iffy on the idea but that would be where my head is at for this.

> Compensation of gas fees

Would be best to disperse on Optimism if gas fees do become a concern. As a side note, if I had to guess I think right now gas fees cost us about 1-2% of the amounts we are paying out right now on average (e.g. I have seen around $1-$2 gas fee for $100 bounty)

> Although, they will pay fee for minting uAD

I think mint should be free, but conditional based on economic conditions like FRAX (e.g. can only mint when above peg) otherwise you need to purchase on the market and we collect our swap fee.

**Reply by sergfeldman** [reply]
> We could provide discounts for bulk buys. We just need to do some rough calculations e.g. we can get 10% APR on $50k so we should give a 5% discount if we think it'll take them a year to spend it...or something like that.

@pavlovcik Providing discounts is a great idea!
It even might be turned into bonds that are sold with a discount.

**Comment by nicholasnokia** [comment]
Was thinking about this
1. Current benefits to holding ubi dollars
Used as payments.
Can be used to obtain yield by depositing in the yield aggregator. 
Qns. How about an automated process by a yield bearing Ubi dollars? Over time , it inflates in value.
https://twitter.com/FroyoFren/status/1613510719932366848

1st step - Users will have to 1st mint usdc - Cash in bank
2nd step - ...  DevPool partners mint Ubiquity Dollars, then thats "cash in the bank" that we should be deploying as part of our normal collateral strategies for example, the straddle (call/put option) plus ETH LP strategy - Revenue split to pay bounties and users who hold ubi dollars.

**Reply by sergfeldman** [reply]
### Develop now - pay later

A potential revenue stream for Ubiquity and a benefit for partners is to provide loans for bounties for the trusted crypto projects.

Trusted crypto projects will get their bounties done without upfront or immediate payment. After some period of time they will repay the loan with additional percentage like for a regular loan.

Ubiquity will pay bounty hunters at once as bounty is done, and get income from the additional percentage.
